src/main/java/net/mtrop/doom/tools/decohack/patches/PatchDHEExtended.java

```diff
@@ -123,6 +327,24 @@
 		"SP99",
 	};
 
+	private static final Map<String, Integer> MAP_SOUNDINDEX = new HashMap<String, Integer>()
+	{
+		private static final long serialVersionUID = -4513058612574767103L;
+		{
+			// the extended sound range expliticly starts at 500, skipping several
+			// indices, so we need to make the output index relative to that.
+			int extstart = 500;
+			int len = SOUNDSTRINGS.length + SOUNDSTRINGSEXTENDED.length;
+			for (int i = 0; i < len; i++)
+			{
+				if (i >= SOUNDSTRINGS.length)
+					put(SOUNDSTRINGSEXTENDED[i - SOUNDSTRINGS.length].toUpperCase(), i - SOUNDSTRINGS.length + extstart - 1);
+				else
+					put(SOUNDSTRINGS[i].toUpperCase(), i);
+			}
+		}
+	};
+
```

Comment:
The conversion to uppercase strings in the code that builds `MAP_SOUNDINDEX` (line 341) can be avoided if the sounds are defined as uppercase to begin with (`SOUNDSTRINGS` was defined as lowercase in `PatchBoom`, but I think that was motivated by my copy-paste laziness rather than consistency with Doom's string table, because as far as I know, you can't change sound names in Boom and higher).

Also, are MBF's additional sounds used in Extended? If so, they also need to be added here, since the function that will search for a valid sound name is overridden in this class. I wonder if there's a better way to do this sort of thing though - `super.getSoundIndex(name)` could probably be called when a sound name isn't found here.

These classes probably need a rewrite - I wrote these in haste, heh.

Reply:
All done -- re-added MBF sound support and pre-uppercased the extended sound names.